src/visualization/clustering/gmm_vis.py

```diff
@@ -0,0 +1,343 @@
+        self.visualization_box.setTitle("Visualization")
+        self.visualization_box_layout = QVBoxLayout(self.visualization_box)
+
+        if self.has_animation:
+            # animation
+            self.animation_box = QGroupBox()
+            self.animation_box.setTitle("Animation")
+            self.animation_box.setFixedWidth(250)
+            self.animation_box_layout = QFormLayout(self.animation_box)
+
+            self.restart_button = QPushButton("Restart")
+            self.restart_button.clicked.connect(partial(self.click_listener, 'restart'))
+            self.run_button = QPushButton("Start animation")
+            self.run_button.clicked.connect(partial(self.click_listener, 'run'))
+            self.interval_box = QSpinBox()
+            self.interval_box.setMinimum(20)
+            self.interval_box.setMaximum(2000)
+            self.interval_box.setValue(200)
+            self.interval_box.setSingleStep(20)
+
+            self.animation_box_layout.addRow(QLabel("Interval time [ms]:"), self.interval_box)
+            self.animation_box_layout.addRow(self.restart_button)
+            self.animation_box_layout.addRow(self.run_button)
+
+            self.left_column_layout.addWidget(self.animation_box, 0)
+
+        self.fig, axes = plt.subplots()
+        self.canvas = GMMCanvas(self.fig, axes, self.has_animation)
+        self.visualization_box_layout.addWidget(self.canvas, 1)
+        self.update_plot()
+
+        if not self.has_animation:
+            self.visualization_box_layout.addStretch()
+
+            # control buttons
+            self.control_buttons_layout = QHBoxLayout()
+            self.left_box = QSpinBox()
+            self.left_box.setMinimum(1)
+            self.right_box = QSpinBox()
+            self.right_box.setMinimum(1)
+            self.left_button = QPushButton("PREV")
+            self.left_button.clicked.connect(partial(self.click_listener, 'prev'))
+            self.right_button = QPushButton("NEXT")
+            self.right_button.clicked.connect(partial(self.click_listener, 'next'))
+            self.step_label = QLabel("STEP: {}".format(self.current_step))
+            self.control_buttons_layout.addWidget(self.left_button)
+            self.control_buttons_layout.addWidget(self.left_box)
+            self.control_buttons_layout.addStretch()
+            self.control_buttons_layout.addWidget(self.step_label)
+            self.control_buttons_layout.addStretch()
+            self.control_buttons_layout.addWidget(self.right_box)
+            self.control_buttons_layout.addWidget(self.right_button)
+
+            self.visualization_box_layout.addLayout(self.control_buttons_layout, 0)
+        else:
+            self.step_label = QLabel("STEP: {}".format(self.current_step))
+            self.visualization_box_layout.addWidget(self.step_label, 0, alignment=Qt.AlignCenter)
+
+        # description
+        description = "Gaussian Mixture Methods algorithm - steps visualization."
```

Comment:
I think, the description should be longer and describe what we see in the visualization.

Reply:
Maybe we can improve descriptions of all algorithms in other PR